src/Language/PureScript/CoreFn/ToJSON.hs

```diff
@@ -22,26 +23,61 @@ import Language.PureScript.Names
 import Language.PureScript.PSString (PSString, decodeString)
 
 literalToJSON :: (a -> Value) -> Literal a -> Value
-literalToJSON _ (NumericLiteral (Left n)) = toJSON ("IntLiteral", n)
-literalToJSON _ (NumericLiteral (Right n)) = toJSON ("NumberLiteral", n)
-literalToJSON _ (StringLiteral s) = toJSON ("StringLiteral", s)
-literalToJSON _ (CharLiteral c) = toJSON ("CharLiteral", c)
-literalToJSON _ (BooleanLiteral b) = toJSON ("BooleanLiteral", b)
-literalToJSON t (ArrayLiteral xs) = toJSON ("ArrayLiteral", map t xs)
-literalToJSON t (ObjectLiteral xs) = toJSON ("ObjectLiteral", recordToJSON t xs)
+literalToJSON _ (NumericLiteral (Left n))
+  = object
```

Comment:
I don't think you need to go through and replace all of this now, but there's an alternative approach which skips over creating the intermediate data structure: https://hackage.haskell.org/package/aeson-1.2.1.0/docs/Data-Aeson.html#g:7

You'd change expressions from

```haskell
object
  [ T.pack "foo" .= foo
  , T.pack "bar" .= bar
  ]
```

to expressions like

```haskell
pairs
  ( T.pack "foo" .= foo
  <> T.pack "bar" .= bar
  )
```

It's supposedly faster and uses less memory. So given that the json blobs have gotten bigger, it might be a change that makes sense.

Reply:
@joneshf I think we'd need to add `ToJSON` instances to make it work (since the right hand side of `.=` expects an instance of `ToJSON`.  This is a bit more work so I'd like to leave it for later.

Reply:
You could just encode to `Value` on the right-hand-side which picks the trivial `ToJSON` instance.

Reply:
@kRITZCREEK you're right, but let's do that in another PR.